[URGENT] Problem with ZIM validation

I have created a ZIM file with latest mwoffliner and node-libzim (so libzim 6.3.0). Rename the file in `.zim`.

[wikisource_fo_all_nopic_2020-11.zim.zip](https://github.com/openzim/libzim/files/5608841/wikisource_fo_all_nopic_2020-11.zim.zip)

If I run zimcheck on the latest libzim/zim-tools git master (but before `libzim_next` merge), then I get an error:
```
$ zimcheck  wikisource_fo_all_nopic_2020-11.zim
[INFO] Checking zim file /home/kelson/code/mwoffliner/mwo-test-1606494150990/wikisource_fo_all_nopic_2020-11.zim
[INFO] Verifying ZIM-file structure integrity...
Title index is not properly sorted:
  #65: A/Forsíða
  #66: A/Forsíða
  [ERROR] ZIM file's low level structure is invalid
[INFO] Avoiding redundant checksum test (already performed by the integrity check).
[INFO] Searching for metadata entries...
[INFO] Searching for Favicon...
[INFO] Searching for main page...
[INFO] Verifying Articles' content...
[INFO] Searching for redundant articles...
  Verifying Similar Articles for redundancies...
[INFO] Overall Test Status: Fail
[INFO] Total time taken by zimcheck: 0 seconds.
```

So, either we newly write corrupted ZIM files or the validation check is buggy. Would like to figure soon the problem and the fix for obvious reasons.

--- a/src/fileimpl.cpp
+++ b/src/fileimpl.cpp
@@ -627,7 +627,7 @@ std::string pseudoTitle(const Dirent& d)
       }
       const auto direntOffset = readOffset(*urlPtrOffsetReader, a);
       const std::shared_ptr<const Dirent> dirent = readDirent(direntOffset);
-      if ( prevDirent && !(pseudoTitle(*prevDirent) < pseudoTitle(*dirent)) )
+      if ( prevDirent && !(pseudoTitle(*prevDirent) <= pseudoTitle(*dirent)) )
       {
         std::cerr << "Title index is not properly sorted:\n"
                   << "  #" << i-1 << ": " << pseudoTitle(*prevDirent) << "\n"

